packages/typespec-client-generator-core/test/types.test.ts

```diff
@@ -184,6 +184,8 @@ describe("typespec-client-generator-core: types", () => {
           azureLocationProperty: string;
           @format("etag")
           etagProperty: string;
+          @format("date")
```

Comment:
this shouldn't not be a thing, this means a string that is with a known pattern. Please don't make anything to have this being a way to specify dates

Reply:
if you want a date you use the `utcDateTime` or `offsetDateTime` type and specify the encoding.

Reply:
Yeah, @tadelesh and I talked about this. Right now, it's feasible. See test case in `autorest.csharp`: https://github.com/Azure/autorest.csharp/blob/00c74a505f8140f7789934c18a95b3749e0ecf80/test/TestProjects/sdk/newprojecttypespec/Azure.NewProject.TypeSpec/Azure.NewProject.TypeSpec.tsp#L265

So, I think we need to discourage people from using it:
- add a linter rule to report error on `@format("date")` and `@format("date-time")`
- What do we do when the TypeSpec spec is used with an OpenAPI emitter? see https://github.com/microsoft/typespec/blob/da99aa955b497c34cd6deb8f873fecd0d583c85b/packages/website/versioned_docs/version-latest/standard-library/reference/js-api/functions/%24format.md?plain=1#L18-L21

Reply:
it is feasible to write but you shouldn't do anything about it. If the user says the type is a string it is a string not a date regardless of what the format says. `@format` is not a type indication is it just a known string pattern.
This is NOT the same as openapi where erverything neeeded to be a primitive type.